### ISSUE TITLE
XSS (javascript execution vulnerability)

### ISSUE THREAD

**Issue by dobryakov** [lynndylanhurley/devise_token_auth#916]
Steps to reproduce:
1. Setup this gem for oauth through social network (vk.com in my case).
2. Change your username to `<script>alert('Hi!');</script>`
3. Logout and try to login through social network.
4. See the "Hi!" alert :)

Reason:

[app/views/devise_token_auth/omniauth_external_window.html.erb](https://github.com/lynndylanhurley/devise_token_auth/blob/master/app/views/devise_token_auth/omniauth_external_window.html.erb#L18)

Workaround:
```
# app/models/user.rb
def username
  CGI::escapeHTML( read_attribute(:username) )
end
```

**Comment by dobryakov** [comment]
Right solution:

`var data = JSON.parse(decodeURIComponent(' <%= URI::escape( @data.to_json ) %> '));`

**Comment by booleanbetrayal** [comment]
thanks for this catch @dobryakov ! Fixed in e6a3b08d1709c3346bc3288f4c6798f12b8b766b
